### PR TITLE
Disco Tie added

### PR DESCRIPTION
**Pull request by CollinCunningham**

### CODE REVIEW

**Comment by ladyada** [comment]
@danh plz take a look at the BLE part

**Comment by TheKitty** [comment]
> Disco_Tie/code.py 7: Trailing whitespace (trailing-whitespace)
> Disco_Tie/code.py 117: Wrong continued indentation (remove 2 spaces).
>                     input_floor, input_ceiling, 0, NUM_PIXELS)
>                   | ^ (bad-continuation)
> Disco_Tie/code.py 138: Bad indentation. Found 7 spaces, expected 8 (bad-indentation)
> Disco_Tie/code.py 147: Wrong continued indentation (remove 2 spaces).
>                     int(user_color[2]/8))
>                   | ^ (bad-continuation)
> Disco_Tie/code.py 149: Wrong continued indentation (remove 3 spaces).
>                     int(user_color[2]/2))
>                  |  ^ (bad-continuation)
> Disco_Tie/code.py 185: Wrong continued indentation (add 24 spaces).
>                     / (in_max - in_min))
>                     ^                       | (bad-continuation)
> Disco_Tie/code.py 81: Redefining name 'pos' from outer scope (line 133) (redefined-outer-name)
> Disco_Tie/code.py 113: Using the global statement (global-statement)
> Disco_Tie/code.py 143: Using the global statement (global-statement)
> Disco_Tie/code.py 144: Using the global statement (global-statement)
> Disco_Tie/code.py 179: Using the global statement (global-statement)
> Disco_Tie/code.py 188: Using the global statement (global-statement)
Mu is good for checking line length and continuations
In the def function, use variable names not the same as in the main program (I agree, it should be ok)
Use of globals triggers it too. If you need them, define an exclude to allow it.
`# pylint: disable=global-statement` - I think that may work

**Comment by ladyada** [comment]
@CollinCunningham do you know how to lint locally

**Comment by TheKitty** [comment]
Kattni did a guide https://learn.adafruit.com/improve-your-code-with-pylint

**Comment by CollinCunningham** [comment]
> Kattni did a guide https://learn.adafruit.com/improve-your-code-with-pylint

lovely - will do. thnx.

**Comment by ladyada** [comment]
@TheKitty going forward, when you reply to PRs that are not linted, give the author a link to the guide when you reply

**Comment by TheKitty** [comment]
Will do @ladyada

**Comment by CollinCunningham** [comment]
cool - all good w linting. @dhalbert let me know of BLE changes needed.

File: Disco_Tie/code.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        for i in range(NUM_PIXELS):
+            pixel_index = (i * 256 // NUM_PIXELS) + j
+            pixels[i] = wheel(pixel_index & 255)
+        pixels.show()
+        time.sleep(wait)
+
+def audio_meter():
+    mic.record(samples, len(samples))
+    magnitude = normalized_rms(samples)
+    global peak
```
Comment (dhalbert) [inline Disco_Tie/code.py]: You can pass peak in and return it rather than changing a global. That is more Pythonic.

File: Disco_Tie/code.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+def larsen_set(index, color):
+    if index < 0:
+        return
+    else:
+        pixels[index] = color
+
+def larsen(wait):
+    global pos
+    global direction
```
Comment (dhalbert) [inline Disco_Tie/code.py]: Here, pass `pos` and `direction` in, and then return them as a tuple. You can do `return (pos, direction)`, and then in the call, do
```
pos, direction = larsen(wait, pos, direction)
```

Comment (CollinCunningham) [reply Disco_Tie/code.py]: Making this change resulted in erratic animation speed – speeding up and slowing down. I left the function mostly unchanged for now.

Comment (dhalbert) [reply Disco_Tie/code.py]: ok, not sure why, but good enuf

**Pull request merged**